kiwi/boot/functions.sh

```diff
@@ -5524,19 +5524,26 @@ function waitForStorageDevice {
     # function to check access on a storage device
     # which could be a whole disk or a partition.
     # the function will wait until the size of the
-    # storage device could be obtained or the check
-    # counter equals 4
+    # storage device could be obtained or the timeout
+    # is reached. Default timeout is 60 seconds, however
+    # it can be set to different value by setting the
+    # DEVICE_TIMEOUT variable in the kernel command
+    # line.
     # ----
     local IFS=$IFS_ORIG
     local device=$1
     local check=0
+    local limit=30
+    if [ -n ${DEVICE_TIMEOUT##*[!0-9]*} ]; then
```

Comment:
To be honest, I'm with bash for some time but never saw the syntax you used :smile: that's why I asked if bash can do the number check in another way than a regexp too